### ISSUE TITLE
How to debug the webpacking part of the webpack sample?

### ISSUE THREAD

**Issue by bobbrow** [Microsoft/vscode-extension-samples#167]
I'm experimenting with the webpack sample because we want to update the C++ extension to use this, but I'm stuck after installing the `vscode-cpptools` node module.  webpack launches ts-loader (I think) to compile the sample, but I get a `TS2307: Cannot find module 'vscode-cpptools'`.  If I run tsc, it has no problem finding it.  I haven't found anything online that fixes my problem so I'd like to set breakpoints in ts-loader to see if I can figure out what is going wrong.  I created a node launch task that runs the webpack script, but none of my breakpoints hit.  Do you have any tips to help debug this?  I'm feeling stuck and have been unable to make progress on this for several hours.

Thanks!

**Comment by bobbrow** [comment]
Oh, I think I might have just solved it.  The sample has this section in the webpack config:

```jsonc
    module: {
        rules: [{
            test: /\.ts$/,
            exclude: /node_modules/,
            use: [{
                loader: 'ts-loader',
                options: {
                    compilerOptions: {
                        "module": "es6" // override `tsconfig.json` so that TypeScript emits native JavaScript modules.
                    }
                }
            }]
        }]
    },
```

It seems to be overriding the sample's tsconfig.json.  Removing `options` appears to fix the webpack compile.  Can you help me understand why the sample is overriding the tsconfig and whether my "fix" is correct?  It seems to run ok.

**Comment by jrieken** [comment]
> Can you help me understand why the sample is overriding the tsconfig and whether my "fix" is correct? It seems to run ok.

Webpack tree shaking only works when emitting native JS modules, that's why we have it in the sample.

**Comment by bobbrow** [comment]
Yeah, I ran into some problem while debugging it yesterday.  I still need to investigate more, but that was the only way I could get it to compile.  Perhaps it's because the `vscode-cpptools` module has a dependency on vscode.d.ts?  I don't know.  We don't have a lot of experience with this stuff.

**Comment by jrieken** [comment]
Hm... the `vscode` dependency should be OK. did you tell web pack that you are targeting node? Like so https://github.com/Microsoft/vscode-extension-samples/blob/master/webpack-sample/webpack.config.js#L14

**Comment by bobbrow** [comment]
I started with the webpack sample and all I did was `npm i vscode-cpptools` and `npm i vscode-jsonrpc`, so yes, it should be targeting node.
![image](https://user-images.githubusercontent.com/12818240/54765020-01edbf00-4bb6-11e9-8b43-897e66a63564.png)

I forked the repo and these are my commits:
* [npm i vscode-cpptools](https://github.com/bobbrow/vscode-extension-samples/commit/5b89df8c097fea02626149fef86d509486433187)
* [npm i vscode-jsonrpc](https://github.com/bobbrow/vscode-extension-samples/commit/4a46785877cda8faa0e61d4ac31af5186410c4f7)

if I run `npm run webpack` I get the TS2307.  tsc works.
